api/users/schemas.py

```diff
@@ -0,0 +1,108 @@
+# -*- coding: utf-8 -*-
+
+def get_user_social_jsonschema():
```

Comment:
@erinspace, @Johnetordoff's PR adds validation schemas for employment and education information and [implements a `from_json` method](https://github.com/CenterForOpenScience/osf.io/pull/8546/files#diff-517a736b64468905bd7818494c33efdaR1) for loading the correct schema. Would you mind mimicking that approach in this PR?